### PR TITLE
Add new main workflow for miRNA reads, add additional alignment parameters to all workflows

### PR DESCRIPTION
**Pull request by wpoehlm**
A new workflow named 'mirna_single.cwl' has been added. This workflow processes single end fastq files, and requires an additional input file (a gtf file containing only miRNA annotations) compared to the other workflows. All other workflows have also been updated to add a set of optional alignment parameters. This PR also fixes the 'options.json' files for all test jobs, such that they have all of the necessary parameters needed to run on a toil cluster.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +104,20 @@ If this argument is not provided, the default value of `2` will be used. This is
...
+* `outFilterMismatchNmax` : Integer specifying the maximum number of mismatches per pair
+* `outFilterMultimapScoreRange` : Integer specifying the score range for multi-mapping alignments
+* `outFilterMultimapNmax` : Integer specifying the maximum number of multiple alignments for a read
+* `outFilterScoreMinOverLread` : Integer specifying the minimum score for an alignment to be reported, normalized to read length 
+* `outFilterMatchNminOverLread` : Integer specifying the minimum number of matched bases for an alignment to be reported, normalized to read length
+* `outFilterMatchNmin` : Integer specifying the minimum number of matched bases for an alignment to be reported
+* `alignSJDBoverhangMin` : Integer specifying the minimum block size for annotated spliced alignments 
+* `alignIntronMax` : Integer specifying the maximum intron size
+
+For further details about these parameters, please refer to the [STAR manual](https://chagall.med.cornell.edu/RNASEQcourse/STARmanual.pdf)
```
Comment (tthyer) [inline README.md:119]: Will they want to set these arguments for anything other than miRNA?

File: jobs/test-UW-mirna/job.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+    "cwl_wf_url": "https://github.com/Sage-Bionetworks-Workflows/dockstore-workflow-rnaseq/tree/addparams",
+    "cwl_args_url": "https://raw.githubusercontent.com/Sage-Bionetworks-Workflows/dockstore-workflow-rnaseq/addparams/jobs/test-UW-mirna/job.json",
```
Comment (tthyer) [inline jobs/test-UW-mirna/job.json]: These two urls will be changed to point to master?

**Review by tthyer — APPROVED** [review]
I wonder whether the README would be easier to read if you collect all the miRNA-specific parts into a separate markdown file and link to it from the README?

Otherwise, I regret all the duplication that is taking place, but it seems like we will just have to wait for 1.2, and hopefully do a major refactor at that time. Is that what you have planned?

File: README.md [hunk 3]
```diff
@@ -101,6 +104,20 @@ If this argument is not provided, the default value of `2` will be used. This is
 
 An example input json file that contains the required inputs and these optional inputs can be found [here](https://raw.githubusercontent.com/Sage-Bionetworks-Workflows/dockstore-workflow-rnaseq/7d64748a3a6d7cc8cfd9f30fc43c1b9bc79b3b3f/jobs/test-paired-fastq/job.json)
 
+In addition, you may optionally specify the following parameters for the STAR alignment:
```
Comment (tthyer) [inline README.md]: do you want to make a comment here that this is typically used for miRNA?

File: bam_paired.cwl [hunk 4]
```diff
@@ -33,6 +33,25 @@ inputs:
     type: string?
   - id: column_number
     type: int?
+  - id: alignEndsType
```
Comment (tthyer) [inline bam_paired.cwl:36]: should these optional ones be added here if they're only going to be used for miRNA?

File: README.md (hunk 1 above)
Comment (wpoehlm) [reply README.md:119]: While these arguments are essential for miRNA data, I think that it's good to have these options for mRNA as well (especially for non-human data)

File: jobs/test-UW-mirna/job.json (hunk 2 above)
Comment (wpoehlm) [reply jobs/test-UW-mirna/job.json]: good catch, I forgot about this folder

File: README.md (hunk 3 above)
Comment (wpoehlm) [reply README.md]: Good idea, I'll note that it is highly recommended to supply these arguments for miRNA

**Comment by wpoehlm** [comment]
> 
> 
> I wonder whether the README would be easier to read if you collect all the miRNA-specific parts into a separate markdown file and link to it from the README?
> 
> Otherwise, I regret all the duplication that is taking place, but it seems like we will just have to wait for 1.2, and hopefully do a major refactor at that time. Is that what you have planned?

Future development that I have planned will involve adding new tools to all of these workflows rather than creating new main workflows, so I'm hoping this will be reasonable to maintain/document. I agree that it would be good to think about how we can reduce redundancy in the long term. However, I do think that having these new parameters available is beneficial even for the non-mirna workflows

**Pull request merged**